### ISSUE TITLE
Add helpers to work with scalars

### ISSUE THREAD

**Issue by robinchrist** [taocpp/sequences#10]
Hi,

It would be great if there were helpers to work with scalars, e.g. if `multiplies` allowed to multiply a sequence with a scalar

Alternatively there could be a `fill` function or similar that allows creating an integer sequence with all elements identical from a scalar and a size

What do you think?

**Comment by d-frey** [comment]
Did you see `scale`? https://github.com/taocpp/sequences#header-taoseqscalehpp

**Comment by robinchrist** [comment]
Nope, I didn't :man_facepalming: Thank you for the hint.

Maybe a note at `multiplies` would be good, something like
"For the scalar version, see `scale`"

**Comment by robinchrist** [comment]
Is there something similar for `plus`? Like `offset` instead of `scale`

**Comment by d-frey** [comment]
I can add the hint when I have some time.

Currently, there is only `scale` as it is the only thing I encountered in practice. The problem with a library like taocpp/Sequences is, that you could easily add a thousand little helpers, so I'd like to restrict it to cases that are actually useful in practice. Do you have an actual use-case for `offset`?

**Comment by robinchrist** [comment]
Yeah, I'm building unshuffle sequences for complex numbers (AoS to SoA), which is basically
`2*Is..., 2*Is + 1 ...` for a standard `index_sequence<Is...>`

**Comment by d-frey** [comment]
OK, I'll need some time as I'm quite busy.

**Comment by robinchrist** [comment]
Yeah no worries, there's no rush

Maybe I'll try to add such a helper myself, shouldn't be too difficult
